src/textual/css/stylesheet.py

```diff
@@ -634,31 +636,14 @@ def update_nodes(self, nodes: Iterable[DOMNode], animate: bool = False) -> None:
             nodes: Nodes to update.
             animate: Enable CSS animation.
         """
-        rules_map = self.rules_map
         apply = self.apply
 
-        def get_rules(node: DOMNode) -> set[RuleSet]:
-            """Get set of rules for the given node."""
-            return {
-                rule
-                for name in rules_map.keys() & node._selector_names
-                for rule in rules_map[name]
-            }
-
         for node in nodes:
-            rules = get_rules(node)
-            if rules:
-                apply(node, limit_rules=rules, animate=animate)
+            apply(node, animate=animate)
             if isinstance(node, Widget) and node.is_scrollable:
                 if node.show_vertical_scrollbar:
-                    rules = get_rules(node.vertical_scrollbar)
-                    if rules:
-                        apply(node.vertical_scrollbar, limit_rules=rules)
+                    apply(node.vertical_scrollbar)
                 if node.show_horizontal_scrollbar:
-                    rules = get_rules(node.horizontal_scrollbar)
-                    if rules:
-                        apply(node.horizontal_scrollbar, limit_rules=rules)
+                    apply(node.horizontal_scrollbar)
                 if node.show_horizontal_scrollbar and node.show_vertical_scrollbar:
-                    rules = get_rules(node.scrollbar_corner)
-                    if rules:
-                        apply(node.scrollbar_corner, limit_rules=rules)
```

Comment:
Is this part of the optimisation process or is it just a simplification of the code because the check `if rules` was somehow superfluous?

Reply:
The `get_rules` pattern is not handled by `apply`. So its redundant here.